cmd/cluster/dynatrace/hcpMgCmd.go

```diff
@@ -0,0 +1,300 @@
+package dynatrace
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+)
+
+func NewCmdHCPMustGather() *cobra.Command {
+	hcpMgCmd := &cobra.Command{
+		Use:               "must-gather <cluster-id>",
+		Aliases:           []string{"mg"},
+		Short:             "Gather all Pod logs and Application event from HCP",
+		Args:              cobra.ExactArgs(1),
+		DisableAutoGenTag: true,
+		Run: func(cmd *cobra.Command, args []string) {
+			err := mustGather(args[0])
+			if err != nil {
+				cmdutil.CheckErr(err)
+			}
+		},
+	}
+
+	hcpMgCmd.Flags().IntVar(&since, "since", 10, "Number of hours (integer) since which to pull logs and events")
+	hcpMgCmd.Flags().IntVar(&tail, "tail", 100, "Last 'n' logs and events to fetch ")
+	hcpMgCmd.Flags().StringVar(&sortOrder, "sort", "desc", "Sort the results by timestamp in either ascending or descending order. Accepted values are 'asc' and 'desc'")
+
+	return hcpMgCmd
+}
+
+func mustGather(clusterID string) (error error) {
+	accessToken, err := getAccessToken()
+	if err != nil {
+		return fmt.Errorf("failed to acquire access token %v", err)
+	}
+
+	clusterInternalID, managementClusterName, DTURL, err := fetchClusterDetails(clusterID)
+	if err != nil {
+		return err
+	}
+	managementClusterInternalID, _, _, err := fetchClusterDetails(managementClusterName)
+	if err != nil {
+		return err
+	}
+
+	clientset, err := getClientsetFromClusterID(managementClusterInternalID)
+	if err != nil {
+		return err
+	}
+
+	hcpNS, err := GetHCPNamespaceFromInternalID(clientset, clusterInternalID)
+	if err != nil {
+		return err
+	}
+
+	fmt.Println(fmt.Sprintf("Using HCP Namespace %v", hcpNS))
+
+	gatherNamespaces := []string{hcpNS, "hypershift", "cert-manager", "redhat-cert-manager-operator"}
```

Comment:
We're probably also forgetting the kubelet namespace, can we check what else a hypershift dump gathers?